examples/plot_metric_examples.py

```diff
@@ -0,0 +1,410 @@
+# Neighborhood Components Analysis
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# 
+# NCA is an extrememly popular metric-learning algorithm, and one of the
+# first few (published back in 2005).
+# 
+# Neighbourhood components analysis aims at "learning" a distance metric
+# by finding a linear transformation of input data such that the average
+# leave-one-out (LOO) classification performance is maximized in the
+# transformed space. The key insight to the algorithm is that a matrix
+# :math:`A` corresponding to the transformation can be found by defining a
+# differentiable objective function for :math:`A`, followed by use of an
+# iterative solver such as conjugate gradient descent. One of the benefits
+# of this algorithm is that the number of classes :math:`k` can be
+# determined as a function of :math:`A`, up to a scalar constant. This use
+# of the algorithm therefore addresses the issue of model selection. Like
+# LMNN, this algorithm does not try to cluster points from the same class in
+# a unique cluster, because it enforces conditions at a local
+# neighborhood scale.
+#
+# You can read more about it in the paper here:
+# `NCA <https://papers.nips.cc/paper/2566-neighbourhood-components-analysis.pdf>`__.
+# 
+
+nca = metric_learn.NCA(max_iter=1000)
+X_nca = nca.fit_transform(X, Y)
+
+plot(X_nca, Y)
+
+
+######################################################################
+# Local Fischer Discriminant Analysis
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# 
+# LFDA is a linear supervised dimensionality reduction method. It is
+# particularly useful when dealing with multimodality, where one ore more
+# classes consist of separate clusters in input space. The core
+# optimization problem of LFDA is solved as a generalized eigenvalue
+# problem. Like LMNN, and NCA, this algorithm does not try to cluster points
+# from the same class in a unique cluster.
+# 
+# Link to paper:
+# `LFDA <http://www.machinelearning.org/proceedings/icml2006/114_Local_Fisher_Discrim.pdf>`__
+# 
+
+lfda = metric_learn.LFDA(k=2, num_dims=2)
+X_lfda = lfda.fit_transform(X, Y)
+
+plot(X_lfda, Y)
+
+
+######################################################################
+# Relative Components Analysis
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# 
+# RCA is another one of the older algorithms. It learns a full rank
+# Mahalanobis distance metric based on a weighted sum of in-class
+# covariance matrices. It applies a global linear transformation to assign
+# large weights to relevant dimensions and low weights to irrelevant
+# dimensions. Those relevant dimensions are estimated using "chunklets",
+# subsets of points that are known to belong to the same class.
+# 
+# Link to paper:
+# `RCA <https://www.aaai.org/Papers/ICML/2003/ICML03-005.pdf>`__
+# 
+
+rca = metric_learn.RCA_Supervised(num_chunks=30, chunk_size=2)
+X_rca = rca.fit_transform(X, Y)
+
+plot(X_rca, Y)
+
+
+######################################################################
+# Manual Constraints
```

Comment:
We do not say explicitly what is meant by "constraint" before so I think this part is quite confusing

Also I think we want to insist on the fact that many metric learning algorithms only need the weak supervision given by constraints, not labels. In many applications, this is easier to obtain.

Reply:
I agree, I'll reformulate this paragraph, let me know what you think

Reply:
I think we can also delete the `num_constraints=200` in all `*_Supervised` algorithms what do you think ?

Reply:
i guess so, unless it increases the computation time too much

Reply:
That's right, I tried and it tooked almost the same time, so I removed them